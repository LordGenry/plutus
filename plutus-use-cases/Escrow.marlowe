contract Escrow {
    Alice = 0x1
    Bob = 0x2
    Carol = 0x3

    Alice commits 450 Ada before block 10 timeout block 100 as escrow {
        when Alice chose as AliceVote and
            (Bob chose as BobVote or Carol chose as CarolVote) before block 90 then {

            if   (AliceVote is 1 and (BobVote is 1 or CarolVote is 1))
              or (BobVote is 1 and CarolVote is 1) then {
                Alice pays Bob 450 Ada before block 100
            } else redeem escrow
        } else redeem escrow
    }
}
