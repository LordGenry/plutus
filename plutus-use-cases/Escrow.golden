[CommitCash (IdentCC 0) (PubKey {getPubKey = 1}) (Value 450) 10 100 (When (AndObs (PersonChoseSomething (IdentChoice 1) (PubKey {getPubKey = 1})) (OrObs (PersonChoseSomething (IdentChoice 2) (PubKey {getPubKey = 2})) (PersonChoseSomething (IdentChoice 3) (PubKey {getPubKey = 3})))) 90 (Choice (OrObs (AndObs (PersonChoseThis (IdentChoice 1) (PubKey {getPubKey = 1}) 1) (OrObs (PersonChoseThis (IdentChoice 2) (PubKey {getPubKey = 2}) 1) (PersonChoseThis (IdentChoice 3) (PubKey {getPubKey = 3}) 1))) (AndObs (PersonChoseThis (IdentChoice 2) (PubKey {getPubKey = 2}) 1) (PersonChoseThis (IdentChoice 3) (PubKey {getPubKey = 3}) 1))) (Pay (IdentPay 1) (PubKey {getPubKey = 1}) (PubKey {getPubKey = 2}) (Value 450) 100 Null) (RedeemCC (IdentCC 0) Null)) (RedeemCC (IdentCC 0) Null)) Null]